src/metrics/MultiValueBenchmark.py

```diff
@@ -26,38 +26,46 @@ def __multi_value_benchmark(
     noise_amount = 0
 
     overlapped_predicted_by_gt = {label: [] for label in plane_gt_dict.keys()}
+    part_overlapped_predicted_by_gt = {label: [] for label in plane_gt_dict.keys()}
 
     for predicted_label, predicted_plane in plane_predicted_dict.items():
         overlapped_gt_planes = []
+        part_overlapped_gt_planes = []
         for gt_label, gt_plane in plane_gt_dict.items():
-            are_well_overlapped = __are_nearly_overlapped(
+            are_well_overlapped, are_part_overlapped = __are_nearly_overlapped(
                 predicted_plane, gt_plane, overlap_threshold
             )
             if are_well_overlapped:
                 overlapped_gt_planes.append(gt_plane)
                 overlapped_predicted_by_gt[gt_label].append(predicted_label)
 
+            if are_part_overlapped:
+                part_overlapped_gt_planes.append(gt_plane)
+                part_overlapped_predicted_by_gt[gt_label].append(predicted_label)
+
         if len(overlapped_gt_planes) > 0:
             correctly_segmented_amount += 1
         else:
             noise_amount += 1
 
-        if len(overlapped_gt_planes) > 1:
+        if len(part_overlapped_gt_planes) > 0:
```

Comment:
if parts are well overlapped,  then they are part overlapped too) so why you change 1 -> 0 ?